applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/harmonic_analysis_strategy.hpp

```diff
@@ -0,0 +1,743 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//   License:        BSD License
+//   Kratos default license: kratos/license.txt
+//
+//   Project Name:        $StructuralMechanicsApplication $
+//   Last modified by:    $Author: quirin.aumann@tum.de   $
+//   Date:                $Date:            August 2017   $
+//   Revision:            $Revision:                0.0   $
+
+#if !defined(KRATOS_HARMONIC_ANALYSIS_STRATEGY )
+#define  KRATOS_HARMONIC_ANALYSIS_STRATEGY
+
+// System includes
+#include<iostream>
+#include<vector>
+#include<iterator>
+
+// External includes
+#include<boost/timer.hpp>
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "includes/ublas_interface.h"
+#include "solving_strategies/strategies/solving_strategy.h"
+
+// Application includes
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Strategy for solving generalized eigenvalue problems.
+template<class TSparseSpace,
+         class TDenseSpace,
+         class TLinearSolver
+         >
+class HarmonicAnalysisStrategy
+    : public SolvingStrategy<TSparseSpace, TDenseSpace, TLinearSolver>
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    KRATOS_CLASS_POINTER_DEFINITION(HarmonicAnalysisStrategy);
+
+    typedef SolvingStrategy<TSparseSpace, TDenseSpace, TLinearSolver> BaseType;
+
+    typedef typename BaseType::TSchemeType::Pointer SchemePointerType;
+
+    typedef typename BaseType::TBuilderAndSolverType::Pointer BuilderAndSolverPointerType;
+
+    typedef typename TDenseSpace::VectorType DenseVectorType;
+
+    typedef typename TDenseSpace::MatrixType DenseMatrixType;
+
+    typedef typename TDenseSpace::MatrixPointerType DenseMatrixPointerType;
+
+    typedef TSparseSpace SparseSpaceType;
+
+    typedef typename TSparseSpace::VectorType SparseVectorType;
+    
+    typedef typename TSparseSpace::VectorPointerType SparseVectorPointerType;
+
+    typedef typename TSparseSpace::MatrixType SparseMatrixType;
+
+    typedef typename TSparseSpace::MatrixPointerType SparseMatrixPointerType;
+
+    typedef std::complex<double> ComplexType;
+
+    typedef boost::numeric::ublas::vector<ComplexType> ComplexVectorType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor.
+    HarmonicAnalysisStrategy(
+        ModelPart& model_part,
+        SchemePointerType pScheme,
```

Comment:
try to follow the style guide at https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide when writing new code. Here the function arguments are not consistent.